src/main/java/org/kainos/ea/auth/JwtGenerator.java

```diff
@@ -0,0 +1,42 @@
+package org.kainos.ea.auth;
+
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.time.Instant;
+import java.util.Date;
+import java.util.UUID;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTCreator.Builder;
+import com.auth0.jwt.algorithms.Algorithm;
+
+public class JwtGenerator {
+
+    public static KeyPair KEYPAIR;
+
+    public JwtGenerator() throws NoSuchAlgorithmException {
+        KEYPAIR = KeyPairGenerator.getInstance("RSA").generateKeyPair();
+    }
+
+    /**
+     * Generates a JWT token for a specified user
+     *
+     * @param username
+     * @return The JWT token as a String
+     */
+    public String generateJwt(String username) {
+
+        Builder tokenBuilder = JWT.create()
```

Comment:
you can add the role to the token either now or as part of the auth ticket

Reply:
Already in progress in the auth ticket.